lib/OAuth.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+namespace Stripe;
+
+abstract class OAuth
+{
+    /**
+     * Generates a URL to Stripe's OAuth form.
+     *
+     * @param array|null $params
+     * @param array|null $opts
+     *
+     * @return string The URL to Stripe's OAuth form.
+     */
+    public static function authorizeUrl($params = null, $opts = null)
+    {
+        if (!$params) {
+            $params = array();
+        }
+
+        $base = ($opts && array_key_exists('connect_base', $opts)) ? $opts['connect_base'] : Stripe::$connectBase;
+
+        $params['client_id'] = self::_getClientId($params);
+        if (!array_key_exists('response_type', $params)) {
+            $params['response_type'] = 'code';
+        }
+        $query = Util\Util::urlEncode($params);
+
+        return $base . '/oauth/authorize?' . $query;
+    }
+
+    /**
+     * Use an authoriztion code to connect an account to your platform and
+     * fetch the user's credentials.
+     *
+     * @param array|null $params
+     * @param array|null $opts
+     *
+     * @return StripeObject Object containing the response from the API.
+     */
+    public static function token($params = null, $opts = null)
+    {
+        $base = ($opts && array_key_exists('connect_base', $opts)) ? $opts['connect_base'] : Stripe::$connectBase;
+        $requestor = new ApiRequestor(null, $base);
+        list($response, $apiKey) = $requestor->request(
+            'post',
+            '/oauth/token',
+            $params,
+            null
+        );
+        return Util\Util::convertToStripeObject($response->json, $opts);
+    }
+
+    /**
+     * Disconnects an account from your platform.
+     *
+     * @param array|null $params
+     * @param array|null $opts
+     *
+     * @return StripeObject Object containing the response from the API.
+     */
+    public static function deauthorize($params = null, $opts = null)
+    {
+        if (!$params) {
+            $params = array();
+        }
+
+        $base = ($opts && array_key_exists('connect_base', $opts)) ? $opts['connect_base'] : Stripe::$connectBase;
+        $requestor = new ApiRequestor(null, $base);
+        $params['client_id'] = self::_getClientId($params);
+        list($response, $apiKey) = $requestor->request(
+            'post',
+            '/oauth/deauthorize',
+            $params,
+            null
+        );
+        return Util\Util::convertToStripeObject($response->json, $opts);
+    }
+
+    private static function _getClientId($params = null)
```

Comment:
I just noticed that in terms of using an underscore prefix for `private` methods, we're doing it about half and half throughout the rest of code right now. I'm not sure if PHP has a general convention, but I'd guess it shouldn't be as important given that it has real visibility modifiers (unlike Python et all). Either way, fine to leave as is.